### PR TITLE
Added nested teams and parent (address issue#1337)

### PR DESCRIPTION
**Pull request by gshefer**
Address [issue#1337](https://github.com/PyGithub/PyGithub/issues/1337)

Example usage:
```python
>>> org = gh.get_organization('testorggshefer')
>>> teams = list(org.get_teams())
>>> teams
[Team(name="a", id=3593732), Team(name="aa", id=3593733), Team(name="aaa", id=3593746), Team(name="ab", id=3593734)]
>>> nested_teams = list(teams[0].get_teams())
>>> nested_teams
[Team(name="aa", id=3593733), Team(name="ab", id=3593734)]
>>> nested_teams[0].parent
Team(name="a", id=3593732)
```

### CODE REVIEW

File: github/Team.py [hunk 1]
```diff
@@ -263,6 +271,18 @@ def edit(
         )
         self._useAttributes(data)
 
+    def get_teams(self):
+        """
+        :calls: `GET /teams/:id/teams <https://developer.github.com/enterprise/2.17/v3/teams/#list-child-teams>`_
```
Comment (s-t-e-v-e-n-k) [inline github/Team.py]: Please use public URLs, not enterprise URLs.

File: github/Team.py [hunk 2]
```diff
@@ -423,3 +444,7 @@ def _useAttributes(self, attributes):
             )
         if "privacy" in attributes:  # pragma no branch
             self._privacy = self._makeStringAttribute(attributes["privacy"])
+        if attributes.get("parent") is not None:  # pragma no branch
```
Comment (s-t-e-v-e-n-k) [inline github/Team.py]: Please follow the existing pattern: ``if "parent" in attributes``.

File: github/Team.py [hunk 3]
```diff
@@ -263,6 +271,18 @@ def edit(
         )
         self._useAttributes(data)
 
+    def get_teams(self):
+        """
+        :calls: `GET /teams/:id/teams <https://developer.github.com/enterprise/2.17/v3/teams/#list-child-teams>`_
+        :rtype: :class:`github.PaginatedList.PaginatedList` of :class:`github.Team.Team`
+        """
+        return github.PaginatedList.PaginatedList(
```
Comment (s-t-e-v-e-n-k) [inline github/Team.py]: Doesn't this request still need a custom Accept header?

Comment (gshefer) [reply github/Team.py]: Not anymore :-)
You can test against some org.
e.g.
```
curl -s -X GET https://api.github.com/orgs/testorggshefer/teams/aa/teams -H "Accept: application/json" -H "Content-Type: application/json" -u username:password
[
  {
    "name": "aaa",
    "id": 3593746,
    "node_id": "MDQ6VGVhbTM1OTM3NDY=",
    "slug": "aaa",
    "description": "aaa",
    "privacy": "closed",
    "url": "https://api.github.com/teams/3593746",
    "html_url": "https://github.com/orgs/testorggshefer/teams/aaa",
    "members_url": "https://api.github.com/teams/3593746/members{/member}",
    "repositories_url": "https://api.github.com/teams/3593746/repos",
    "permission": "pull",
    "parent": {
      "name": "aa",
      "id": 3593733,
      "node_id": "MDQ6VGVhbTM1OTM3MzM=",
      "slug": "aa",
      "description": "aa",
      "privacy": "closed",
      "url": "https://api.github.com/teams/3593733",
      "html_url": "https://github.com/orgs/testorggshefer/teams/aa",
      "members_url": "https://api.github.com/teams/3593733/members{/member}",
      "repositories_url": "https://api.github.com/teams/3593733/repos",
      "permission": "pull"
    }
  }
]

```

**Pull request merged**